### PR TITLE
Add S3 presign url

### PR DESCRIPTION
**Pull request by tylersouthwick**
* **Please check if the PR fulfills these requirements**
- [x] The commit message describes your change
- [x] Tests for the changes have been added if possible (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [x] Changes are mentioned in the changelog (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
adds s3 presign url support

* **What is the current behavior?** (You can also link to an open issue here)
feature not supported

* **What is the new behavior (if this is a feature change)?**
adds s3PresignURL

* **Does this PR introduce a breaking change?** (What changes might users need to make in their setup due to this PR?)
no

* **Other information**:

### CODE REVIEW

File: src/main/java/de/taimos/pipeline/aws/AwsSdkResponseToJson.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ * Jenkins doesn't support returning the API response directly.
+ * Converting it into a map/list construct allows the full API response to be returned to be used in the jenkinsfile.
+ */
+public class AwsSdkResponseToJson {
```
Comment (hoegertn) [inline src/main/java/de/taimos/pipeline/aws/AwsSdkResponseToJson.java]: Why is this needed in the presign step?

File: src/main/java/de/taimos/pipeline/aws/S3PresignUrlStep.java [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+			return "s3PresignURL";
+		}
+
+		@Override
+		public String getDisplayName() {
+			return "Presign file in S3";
+		}
+	}
+
+	public static class Execution extends StepExecution {
```
Comment (hoegertn) [inline src/main/java/de/taimos/pipeline/aws/S3PresignUrlStep.java]: Please use a `SynchronousNonBlockingStepExecution` here to make the code easier.

Comment (tylersouthwick) [reply src/main/java/de/taimos/pipeline/aws/S3PresignUrlStep.java]: Cool. didn't know about that implementation

File: src/main/java/de/taimos/pipeline/aws/AwsSdkResponseToJson.java (hunk 1 above)
Comment (tylersouthwick) [reply src/main/java/de/taimos/pipeline/aws/AwsSdkResponseToJson.java]: its not. I checked it out from a different branch (not master)

**Comment by tylersouthwick** [comment]
added expiration and httpMethod and rebased off of master. I also removed the AwsSdkResponseToJson from this PR. It now uses `SynchronousNonBlockingStepExecution`.
